pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,300 @@
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevelConfigurationList is a list of PriorityLevelConfiguration objects.
+type PriorityLevelConfigurationList struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta
+	// Items is a list of request-priorities.
+	Items []PriorityLevelConfiguration
+}
+
+type PriorityLevelConfigurationSpec struct {
+	// Exempted defines whether the priority level is exempted or not.  There should be at most one exempt priority level.
+	// Being exempt means that requests of that priority are not subject to concurrency limits (and thus are never queued)
+	// and do not detract from the concurrency available for non-exempt requests. In a more sophisticated system, the
+	// exempt priority level would be the highest priority level.
+	// +optional
+	Exempted *bool
+	// AssuredConcurrencyShares is a number representing the weight by which the priority level shares the concurrency
+	// from the global limit.The concurrency limit of an apiserver is divided among the non-exempt priority levels in
+	// proportion to their assured concurrency shares. Basically this produces the assured concurrency value (ACV) for each non-exempt
+	// priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	AssuredConcurrencyShares int32
+	// Queues is a number of queues that belong to a non-exempt PriorityLevelConfiguration object. The queues exist independently
+	// at each apiserver.
+	Queues int32
+	// HandSize is a small positive number for applying shuffle sharding. When a request arrives at an apiserver the
+	// request flow identifier’s string pair is hashed and the hash value is used to shuffle the queue indices and deal
+	// a hand of size. If empty, the hand size will the be set to 1 at which  the priority drains the queues on a FIFO basis.
+	// +optional
+	HandSize int32
+	// QueueLengthLimit is a length limit applied to each queue belongs to the priority.
+	// +optional
+	QueueLengthLimit int32
+}
+
+// PriorityLevelConfigurationStatusConditionType is a valid value for PriorityLevelConfigurationStatusCondition.Type
+type PriorityLevelConfigurationConditionType string
+
+// PriorityLevelConfigurationStatusConditionType is a valid value for PriorityLevelConfigurationStatusCondition.Status
+type PriorityLevelConfigurationConditionStatus string
+
+// PriorityLevelConfigurationStatus represents the current state of a request-priority.
+type PriorityLevelConfigurationStatus struct {
+	// CurrentQueues is the current number of existing queues that belong to this non-exempt PriorityLevelConfiguration object.
+	CurrentQueues *int32
+	// CurrentAssuredConcurrencyShares is the current number of assured concurrency divided from the system global limit.
+	CurrentAssuredConcurrencyShares *int32
```

Comment:
This, too, can potentially vary from one apiserver to another.

Reply:
removed the field